src/backend/distributed/test/distributed_transaction_id.c

```diff
@@ -0,0 +1,167 @@
+/*-------------------------------------------------------------------------
+ *
+ * test/src/distributed_transaction_id.c
+ *
+ * This file contains functions to exercise distributed transaction id
+ * functionality within Citus.
+ *
+ * Copyright (c) 2017, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+#include "postgres.h"
+#include "c.h"
+#include "fmgr.h"
+#include "funcapi.h"
+#include "miscadmin.h"
+
+#include "catalog/pg_type.h"
+#include "distributed/distributed_transaction_management.h"
+#include "distributed/metadata_cache.h"
+#include "distributed/listutils.h"
+#include "distributed/test_helper_functions.h" /* IWYU pragma: keep */
+#include "nodes/pg_list.h"
+#include "utils/timestamp.h"
+
+
+static int CompareDistributedTransactionIds(const void *leftElement,
+											const void *rightElement);
+
+
+PG_FUNCTION_INFO_V1(get_all_active_distributed_transaction_ids);
+
+
+/*
+ * get_all_active_distributed_transaction_ids returns all the
+ * active distributed transaction ids.
+ */
+Datum
+get_all_active_distributed_transaction_ids(PG_FUNCTION_ARGS)
```

Comment:
I assume you're going to to that anyway, but to be sure: If you move, it it shouldn't be in test/* ;)